### ISSUE TITLE
Fix Request: false positive in OptimizeMapOrSetForEnum

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#573]
```
import java.util.HashMap;
import java.util.Map;

public class MapSetEnumIssue {

    private static final Map<String, String> map;

    static {
        map = new HashMap<>(); // <- false positive
    }
}
```

**Comment by jborgers** [comment]
PR merged
